### ISSUE TITLE
[MDEP-653] add info message to purge-local-repository goal even in non-verbose mode

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-dependency-plugin#1088]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** opened **[MDEP-653](https://issues.apache.org/jira/browse/MDEP-653?redirect=false)** and commented

currently, there is no info message apart from download status while re-resolving (or in verbose mode, there is too much output):

```
$ mvn dependency:purge-local-repository
[INFO] Scanning for projects...
[INFO] 
[INFO] ----------< org.apache.maven.plugins:maven-dependency-plugin >----------
[INFO] Building Apache Maven Dependency Plugin 3.1.2-SNAPSHOT
[INFO] ----------------------------[ maven-plugin ]----------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:3.1.1:purge-local-repository (default-cli) @ maven-dependency-plugin ---
Downloading from central: http://localhost:8081/nexus/content/groups/public/org/apache/maven/maven-artifact/3.0/maven-artifact-3.0.pom
Downloaded from central: http://localhost:8081/nexus/content/groups/public/org/apache/maven/maven-artifact/3.0/maven-artifact-3.0.pom (1.9 kB at 29 kB/s)
[...]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
```

would be useful to write something like:

```
[INFO] Deleting 75 projects' transitive dependencies from /home/herve/.m2/repository at artifact version level
[INFO] Re-resolving dependencies
Downloading from central: http://localhost:8081/nexus/content/groups/public/org/apache/maven/maven-artifact/3.0/maven-artifact-3.0.pom
Downloaded from central: http://localhost:8081/nexus/content/groups/public/org/apache/maven/maven-artifact/3.0/maven-artifact-3.0.pom (1.9 kB at 29 kB/s)
[...]
```

with bold color for configurable arguments (count, "transitive" vs "direct", "version" or other fuzzyness values)

---

**Affects:** 3.1.1

**Remote Links:**
- [GitHub Pull Request #14
](https://github.com/apache/maven-dependency-plugin/pull/14)
- [GitHub Pull Request #15
](https://github.com/apache/maven-dependency-plugin/pull/15)
- [GitHub Pull Request #16
](https://github.com/apache/maven-dependency-plugin/pull/16)

**Comment by jira-importer** [comment]
**[Pim Moerenhout](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=pmoerenhout)** commented

After reviewing, I made some minor improvements.

See https://github.com/apache/maven-dependency-plugin/pull/15

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

reviewing once more:
- what about multi-module purge?
- what about manual purge (instead of project dependencies)?
- "Re-resolving dependencies" message is missing
- are some debug messages not redundant with the new ones?

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

thanks Pim

In fact, as often with Maven plugins, there is the initial idea that we have on one usual case, then when you look at the plugin code, you see that it has code to deal with cases you didn't think at: you learn details about the plugin.
The idea is just about testing in real life these cases and seeing if there are some improvements to be done to better handle them, consistently with the initial usual case.

Then I'm just citing the cases I saw in the code:
- multi-modules: there is some code for multi-module purge, then I suppose the message will be written on each module: in such case, writing down the artifact name could give a better user experience than many times "project's dependencies"
- manual purge: see https://maven.apache.org/plugins/maven-dependency-plugin/examples/purging-local-repository.html#Manual_purge, the output should be improved like it was done with projects dependencies purge
- "Re-resolving dependencies" message is missing: this one was just forgotten from the initial patch. The message is useful for the user to understand why purging downloads artifacts (which is not obvious at first)
- redundancy with debug messages: the improvement is for use in default non verbose mode. But in verbose mode, I saw at least one message "Purging dependencies for project: " that become redundant and should be removed. Perhaps there are others

everything remains quite simple IMHO, just has more cases to check than what is written in the description: welcome to Maven plugins world, where simple does never mean super simple :)
